Hw 262 new mailing schedule step

meladawy opened the pull request:
Here is the changes : 
1- Define Wizard as a new crm component 
2- Change placeholder color for text fields 
3- Add 2px rounded corner for panels as @jamienovick suggestion

![hw-262](https://cloud.githubusercontent.com/assets/2423218/24503336/ede893b4-1551-11e7-8ce6-4a40b4c37cb5.png)


Wizard Component HTMl Structure 

```html
<div class="crm_wizard">

    <div class="panel crm_wizard__title">
        <div class="panel-body">
            <ul class="nav nav-pills">
                <li class="active">
                    <a href="">
                        <span class="crm_wizard__title__number"> 1 </span>
                        Step 1
                    </a>
                </li>
                
                <li>
                    <a href="">
                        <span class="crm_wizard__title__number"> 2 </span> 
                        Step 2
                    </a>
                </li>
                
                <li>
                    <a href="">
                        <span class="crm_wizard__title__number"> 3 </span> 
                        Step 3
                    </a>
                </li>
            </ul>
        </div>
    </div>

    <div class="panel crm_wizard__body">
        <div class="panel-body">
            Any body content
        </div>
    </div>

    <div class="panel crm_wizard__footer">
        <div class="panel-body">
            <div class="row">
                <div class="col-md-6 text-left">
                    <button class="btn-danger btn">Action</button>
                </div>
                <div class="col-md-6 text-right">
                    <button class="btn-primary btn">Action</button>
                </div>
            </div>
        </div>
    </div>

</div>
```

Wizard Component Output

![screen shot 2017-03-30 at 3 15 16 pm](https://cloud.githubusercontent.com/assets/2423218/24505839/bd67f4be-155b-11e7-9193-59c1fbfc0467.png)

AkA84 commented on scss/bootstrap/overrides/style/_panel.scss:
```diff
@@ -1,8 +1,10 @@
 .panel {
   @include box-shadow($crm-panels-box-shadow);
 
+  border-radius: 2px;
```
in bootstrap i think there is a generic border-radius-related variable, shouldn't we use that?

AkA84 commented on scss/bootstrap/overrides/style/_forms.scss:
```diff
@@ -25,6 +25,30 @@ label {
   }
+
+  &[type="text"] {
+    &::-webkit-input-placeholder {
```
What's the need for all these browser-specific styles?

meladawy replied:
Shoudln't we support all the browsers?

AkA84 replied:
apologies, i just realized that this being for the placeholder it requires vendor-specific selectors

AkA84 commented on scss/bootstrap/overrides/style/_panel.scss:
```diff
@@ -1,8 +1,10 @@
+  border-radius: 2px;
   border: none;
   margin-bottom: $grid-gutter-width;
+  overflow: hidden;
```
Why is this needed?

meladawy replied:
Border radius is not applied to panels without having overflow hidden. not sure if there is any better approach

AkA84 commented on scss/bootstrap/components/_wizard.scss:
```diff
@@ -1,11 +1,11 @@
+.crm_wizard {
+  &__body,
+  &__title {
     margin-bottom: 30px;
```
Does this mean that any `.crm_wizard` will have a bottom margin?

meladawy replied:
Yea this margin should be applied to any wizard. 

![wizard_margin](https://cloud.githubusercontent.com/assets/2423218/24503935/7b682784-1554-11e7-88bc-e26b4ff25dc1.png)

meladawy commented on scss/bootstrap/overrides/style/_panel.scss:
```diff
@@ -1,8 +1,9 @@
+  border-radius: $border-radius-small;
   border: none;
   margin-bottom: $grid-gutter-width;
+  overflow: hidden;
```
This overflow property required to get border-radius working...not sure if there is any better approach

AkA84 replied:
yeah i think we need to discuss this

meladawy replied:
I removed the overflow hidden and replaced it with another border-radius in .panel-body

plz check this commit
https://github.com/compucorp/org.civicrm.shoreditch/pull/5/commits/2f1e119b5635437b3866ff1717d11edc78c27fca

AkA84 commented on scss/bootstrap/overrides/style/_forms.scss:
```diff
@@ -25,6 +25,30 @@ label {
+    }
+
+    &:-moz-placeholder {
+      /* Mozilla Firefox 4 to 18 */
```
this is not necessary, firefox is currently at v52

meladawy replied:
Resolved 
https://github.com/compucorp/org.civicrm.shoreditch/pull/5/commits/a9b01d5fbb541333ceb876873f3544afcbfac9d2

AkA84 commented on scss/bootstrap/overrides/style/_forms.scss:
```diff
@@ -25,6 +25,30 @@ label {
+  &[type="text"] {
+    &::-webkit-input-placeholder {
+      /* WebKit, Blink, Edge */
+      color: $crm-dark-grayblue;
```
Remove all these comments, it's evident enough what browser each selector is for, it's in the prefix

meladawy replied:
Resolved 

https://github.com/compucorp/org.civicrm.shoreditch/pull/5/commits/a9b01d5fbb541333ceb876873f3544afcbfac9d2

meladawy commented:
@AkA84 I removed radius from panels and added it to Wizard as per our yesterday discussion 
https://github.com/compucorp/org.civicrm.shoreditch/pull/5/commits/25198903dbd06147ffef1ab32d652eb6e88cf72a
